autosubmit/git/autosubmit_git.py

```diff
@@ -301,3 +303,19 @@ def check_unpushed_changes(expid: str) -> None:
                         # M: Modified, A: Added, D: Deleted, ?: Untracked
                         raise AutosubmitCritical("Push local changes to remote repository before running", 7075)
 
+    @staticmethod
+    def check_directory_in_use(expid: str) -> bool:
+        """
+        Checks for open files and unfnished git-credential requests in a directory
+        """
+        if process_id(expid) is not None:
+            return True
+        for proc in psutil.process_iter(['pid', 'name', 'cmdline']):
+            name = proc.info.get('name', '')
+            cmdline = proc.info.get('cmdline') or []
+            if 'git-credential' in name or any('git-credential' in arg for arg in cmdline):
+                return True
```

Comment:
Do we need to check if `git-credential` is running? If we have 30 experiments with a lot of users connecting, like in the ClimateDT VM, couldn't we have multiple `git-credential` running, from different folders and for different users?

I thought maybe checking just if the `autosubmit run` was active would be enough?